### PR TITLE
Partially outline code inside the panic! macro

### PR DESCRIPTION
**Pull request by Zoxc**
This outlines code inside the panic! macro in some cases. This is split out from https://github.com/rust-lang/rust/pull/115562 to exclude changes to rustc.

### CODE REVIEW

**Comment by Zoxc** [comment]
@RalfJung I applied your attribute suggestion here.

File: library/core/src/panicking.rs [hunk 1]
```diff
@@ -161,8 +169,10 @@ pub fn unreachable_display<T: fmt::Display>(x: &T) -> ! {
 
 #[inline]
 #[track_caller]
-#[lang = "panic_display"] // needed for const-evaluated panics
```
Comment (RalfJung) [inline library/core/src/panicking.rs:164]: This looks like you can remove the panic_display lang item entirely?

**Comment by m-ou-se** [comment]
@Zoxc Did this actually improve anything? This PR adds no tests or benchmarks, and the perf results above show no improvements.

**Comment by Zoxc** [comment]
It does improve the code generation as stated. You'd probably need some rather hot affected code to measure anything though. Inlining would probably be mostly effected as this reduces the code size of panics.

**Pull request merged**